crates/odrl/src/model/party.rs

```diff
@@ -27,16 +43,19 @@ pub struct Party {
     pub part_of: Vec<PartyCollection>,
     #[serde(skip_serializing)]
     pub function: Function,
+    #[serde(rename = "@type", skip_serializing_if = "Option::is_none")]
+    pub r#type: Option<PartyType>,
 
 }
 
 impl Party {
 
-    pub fn new(uid: Option<IRI>, part_of: Vec<PartyCollection>, function: Function) -> Party {
+    pub fn new(uid: Option<IRI>, part_of: Vec<PartyCollection>, function: Function, r#type: Option<PartyType>) -> Party {
```

Comment:
Can you call this party_type or something like this ?
I really hate the r#type syntax since it is very error prone and not really ergonomic to use.